Feat/get sitable pose

vitoranello opened the pull request:
A service thats return a sitable_position

vitoranello committed:
Refact pose_writer to remove ros__parameters

vitoranello committed:
Add a service thats returns a sitable_position

kriskappel commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -42,18 +42,19 @@ def __init__(self, node_name: str = 'pose'):
+  def readPose(self, dict: str, key: str):
     """
     @brief Reads pose data for a given key from the Redis database.
     @param key: The key identifying the target.
```
missin dict in doc

vitoranello replied:
Ty :)

kriskappel commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -97,14 +98,41 @@ def setStaticPose(self):
+          pipe.hmset(key, pose)
       pipe.execute()
+
+  def getSitablePose(self, req : GetPose.Request, res : GetPose.Response):
```
why are you receiving the response if you are creating an obj from that type within the func

vitoranello replied:
YE, i removed

kriskappel reviewed (approved):
some inconsistencies that have to be addressed

richassis commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -42,18 +42,19 @@ def __init__(self, node_name: str = 'pose'):
     self.setStaticPose()
-    self.pose_server = self.create_service(GetPose, '/fbot_world/get_pose', self.getPose)
+    self.pose_server = self.create_service(GetPose, '/fbot_world/get_pose', self.getTargets)
+    self.sitable_server = self.create_service(GetPose, '/fbot_world/get_sitable_pose', self.getSitablePose)
```
```suggestion
```
Use only one service

richassis commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -97,14 +98,41 @@ def setStaticPose(self):
+    self.get_logger().info("Targets request")
+    res = self.getPose(dict = 'sitable_positions', key = req.key)
+    return res
+
```
```suggestion
```
Use only one service

richassis commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -97,14 +98,41 @@ def setStaticPose(self):
+    return res
   
-  def getPose(self, req : GetPose.Request, res : GetPose.Response):
+  def getPose(self, dict: str, key: str):
```
```suggestion
  def getPose(self, req : GetPose.Request, res : GetPose.Response):
```
The function will now be the service callback.

richassis commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -97,14 +98,41 @@ def setStaticPose(self):
+    res = GetPose.Response()
+    self.get_logger().info("Targets request")
+    res = self.getPose(dict = 'targets', key = req.key)
+    return res
```
```suggestion
```
Not needed anymore.

richassis commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -42,18 +42,19 @@ def __init__(self, node_name: str = 'pose'):
 
     self.setStaticPose()
-    self.pose_server = self.create_service(GetPose, '/fbot_world/get_pose', self.getPose)
+    self.pose_server = self.create_service(GetPose, '/fbot_world/get_pose', self.getTargets)
```
```suggestion
    self.pose_server = self.create_service(GetPose, '/fbot_world/get_pose', self.getPose)
```
No need for an intermediate function anymore.

richassis commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -119,21 +147,21 @@ def getPose(self, req : GetPose.Request, res : GetPose.Response):
+      rclpy.logging.get_logger('pose_plugin').error("Key is empty: " + str(key))
       res.error = 1
       return res
-    if req.key not in self.targets.keys():
-      rclpy.logging.get_logger('pose_plugin').error("Key not found in targets: " + str(req.key))
+    if key not in self.targets[dict].keys():
+      rclpy.logging.get_logger('pose_plugin').error("Key not found in "+dict+": " + str(key))
       res.error = 2
       return res
     res.error = 0
-    key = req.key
-    pose = self.readPose(key)
+    key = key
+    pose = self.readPose(dict, key)
     res.pose = pose
     rclpy.logging.get_logger('pose_plugin').info("Pose: " + str(pose))
-    size = self.readSize(key)
+    size = self.readSize(dict, key)
     res.size = size
     rclpy.logging.get_logger('pose_plugin').info("Size: " + str(size))
     return res
```
The key now will be given by "req.key", and also the "dict" value (now called class) will be given by "req.class" field (stored in "class_name" variable in my previous comment).

richassis commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -119,21 +147,21 @@ def getPose(self, req : GetPose.Request, res : GetPose.Response):
     '''
 
     res = GetPose.Response()
```
```suggestion
```
Not needed anymore.

richassis commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -119,21 +147,21 @@ def getPose(self, req : GetPose.Request, res : GetPose.Response):
     '''
 
```
```suggestion

class_name = req.class if req.class != 'None' else 'targets'
```
Add the default value "targets" as an alternative in case no class was set to the request.

vitoranello committed:
Transform two services in one.
Now you need call with area and key.
area: targets, sitable_poses, etc ...
key: left_chair, exit, living_room, etc ...

If you dont one argument, area is define as 'targets' and key is define the argumet.

vitoranello committed:
Rename dict to area and add brief

vitoranello committed:
Rename area to class_name

vitoranello committed:
Add class_name in GetPose.srv

richassis reviewed (approved):
You're the man!

Jfsslemos commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -45,15 +44,16 @@ def __init__(self, node_name: str = 'pose'):
     self.get_logger().info(f"Pose node started!!!")
 
-  def readPose(self, key: str):
+  def readPose(self, class_name: str, key: str):
```
```suggestion
  def readPose(self, group_set: str, key: str):
```

Jfsslemos commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -97,45 +98,59 @@ def setStaticPose(self):
-      - 2: Key not found in locations
+      - 1: Class and key is empty
+      - 2: Class not found in targets
+      - 3: Key not found in the specified class
```
if there is an error, atribute an invalid value for the rest of the fields

Jfsslemos reviewed (approved):
Inspirational work

vitoranello committed:
Class_name changed to group_set in pose.py and in services msgs
GetPoseFromClass name changed to GetPoseFromSet
Add srv and msg in CMake
Now key cannot be empty. But if group_set is empty it will be set to 'targets'

vitoranello committed:
Now the service return pose and size as -999999.999

vitoranello committed:
Added a function that sets POSE and SIZE to 'NaN' when an error occurs
